app/server_app_adapters.go

```diff
@@ -62,10 +63,17 @@ func (s *Server) RunOldAppInitialization() error {
 	if s.FakeApp().Srv.newStore == nil {
 		s.FakeApp().Srv.newStore = func() store.Store {
 			return store.NewTimerLayer(
-				localcachelayer.NewLocalCacheLayer(
-					sqlstore.NewSqlSupplier(s.FakeApp().Config().SqlSettings, s.Metrics),
-					s.Metrics, s.Cluster, s.CacheProvider),
-				s.Metrics)
+				searchlayer.NewSearchLayer(
```

Comment:
I have one question. I've seen that this works as a cascade going down in the layers but I can't find how this works when a method is missing. For example:

You call `a.Srv.Store.Webhook()`, `TimeLayer` is called and the WebHook method is called in that layer. Inside the code that layer is going to call `s.WebhookStore.Method()` and that call goes to the `SearchLayer` which doesn't have `Webhook` method defined. Am I missing something or that is going to fail?

Reply:
You are missing something, we are using "Structure embedding", which is kinda "inheritance" but based on composition. You can learn more about that in this talk https://www.youtube.com/watch?v=-LzYjMzfGDQ

Reply:
Yeah I know about "Structure embedding" but I didn't see the `store.Store` at first in the `SearchStore` struct. Great thank you! (I'll watch the talk anyways hehehe)